File: 剑指offer/一刷/面试题10- II. 青蛙跳台阶问题.py
class Solution(object):
    def numWays(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 0:
            return 1
        elif n == 1:
            return 1
        elif n == 2:
            return 2
        else:
            res = [0]*(n+1)
            res[0] = 0
            res[1] = 1
            res[2] = 2

            for i in range(3,n+1):
                res[i] = res[i-1] + res[i-2]

            return res[-1] % 1000000007





if __name__ == "__main__":
    solution = Solution()
    solution.jumpFloor(3)

File: 剑指offer/面试题10- II. 青蛙跳台阶问题.go
package 剑指offer


func numWays(n int) int {

	if n == 0 {
		return 1
	}

	if n <= 2 {
		return n
	}

	// go 创建数组的正确方式
	res := make([]int, n+1)
	res[0] = 1
	res[1] = 1

	// go 做for循环的正确方式
	for i := 2; i < n+1; i++ {
		res[i] = (res[i-1] + res[i-2]) % 1000000007
	}
	return res[n]


}